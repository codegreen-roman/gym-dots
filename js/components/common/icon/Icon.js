import React from 'react'
import { string } from 'prop-types'
import { svgStyle, pathStyle } from './Icon.glamor'

const icons = {
    menu:'M14 17h-10v2h10v-2zm6-8h-16v2h16v-2zm-16 6h16v-2h-16v2zm0-10v2h16v-2h-16z',
    facebook:'M6.491 19.991v-9.117H9.44l.442-3.557h-3.39V5.049c0-1.033.272-1.73 1.7-1.73H10V.14C9.686.097 8.607 0 7.358 0 4.74 0 2.957 1.66 2.957 4.695v2.622H0v3.557h2.957V20H6.49v-.009z',
    twitter:'M20 1.911a8.23 8.23 0 0 1-2.356.643A4.1 4.1 0 0 0 19.448.298a8.3 8.3 0 0 1-2.607.99A4.102 4.102 0 0 0 13.846 0C11.58 0 9.743 1.827 9.743 4.079c0 .319.036.63.106.93A11.67 11.67 0 0 1 1.393.745a4.072 4.072 0 0 0 1.27 5.446 4.12 4.12 0 0 1-1.858-.513v.051c0 1.976 1.415 3.625 3.29 4a4.178 4.178 0 0 1-1.08.143 3.96 3.96 0 0 1-.773-.076 4.102 4.102 0 0 0 3.832 2.834 8.261 8.261 0 0 1-5.095 1.743c-.332 0-.658-.02-.979-.056a11.645 11.645 0 0 0 6.289 1.836c7.547 0 11.673-6.216 11.673-11.607l-.014-.528A8.168 8.168 0 0 0 20 1.91',
    guest: 'M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z',
    dumbbells:'M33.48 6.796c-.353-.444-.85-.698-1.364-.698h-.773v-.653c0-1.173-.973-2.128-2.169-2.128h-1.018V2.112a2.15 2.15 0 0 0-.633-1.519c-.8-.79-2.183-.79-2.983 0a2.151 2.151 0 0 0-.634 1.52v5.654H10.094V2.113A2.15 2.15 0 0 0 9.46.594c-.8-.791-2.183-.791-2.983 0a2.149 2.149 0 0 0-.633 1.519v1.205H4.7a2.06 2.06 0 0 0-1.43.571c-.39.376-.615.9-.615 1.437v.772H1.89c-.516 0-1.016.255-1.37.7A2.516 2.516 0 0 0 0 8.362v3.257c0 .585.185 1.14.522 1.563.354.446.853.701 1.37.701h.764v.772c0 .537.224 1.06.616 1.437.385.368.892.571 1.43.571h1.142v1.205c0 .568.23 1.121.633 1.519.4.395.93.613 1.492.613a2.11 2.11 0 0 0 1.491-.613 2.15 2.15 0 0 0 .634-1.519v-5.653h13.812v5.653c0 .568.231 1.121.633 1.519.4.395.93.613 1.492.613a2.11 2.11 0 0 0 1.492-.613 2.15 2.15 0 0 0 .633-1.519v-1.205h1.018c1.196 0 2.17-.954 2.17-2.127v-.653h.772c.513 0 1.01-.255 1.364-.698a2.5 2.5 0 0 0 .52-1.557V8.353c0-.583-.185-1.136-.52-1.557zM1.89 12.77c-.202 0-.399-.107-.553-.3a1.384 1.384 0 0 1-.276-.852V8.362c0-.321.1-.632.276-.852.154-.193.35-.3.553-.3h.765v5.56H1.89zm2.81 2.78c-.267 0-.527-.102-.71-.278a.854.854 0 0 1-.272-.618V5.326c0-.23.096-.45.27-.617.186-.178.445-.28.712-.28h1.143v11.122H4.7zm4.33 2.317a.993.993 0 0 1-.3.71c-.398.394-1.126.394-1.524 0a.992.992 0 0 1-.3-.71V2.113c0-.267.106-.52.3-.71a1.091 1.091 0 0 1 1.524 0c.193.19.3.443.3.71v15.756zm1.063-6.765V8.878h13.812v2.225H10.094zm17 6.765a.993.993 0 0 1-.3.71c-.4.394-1.127.394-1.525 0a.992.992 0 0 1-.3-.71V2.112a.99.99 0 0 1 .3-.709 1.091 1.091 0 0 1 1.524 0c.194.19.3.443.3.71v15.755zm3.187-3.332c0 .56-.496 1.015-1.107 1.015h-1.018V4.43h1.018c.61 0 1.107.456 1.107 1.016v9.091zm2.656-2.909c0 .32-.1.627-.273.846-.153.192-.348.298-.548.298h-.773V7.21h.773c.2 0 .395.106.548.298.174.218.273.526.273.845v3.274z',
    check:'M13.5 0c1.863 0 3.617.352 5.26 1.055a13.547 13.547 0 0 1 4.298 2.887 13.547 13.547 0 0 1 2.887 4.298C26.648 9.883 27 11.637 27 13.5s-.352 3.617-1.055 5.26a13.547 13.547 0 0 1-2.887 4.298 13.547 13.547 0 0 1-4.298 2.887C17.117 26.648 15.363 27 13.5 27s-3.617-.352-5.26-1.055a13.547 13.547 0 0 1-4.298-2.887 13.547 13.547 0 0 1-2.887-4.298C.352 17.117 0 15.363 0 13.5s.352-3.617 1.055-5.26a13.547 13.547 0 0 1 2.887-4.298A13.547 13.547 0 0 1 8.24 1.055C9.883.352 11.637 0 13.5 0zm0 25.339c1.635 0 3.168-.312 4.601-.936a11.921 11.921 0 0 0 3.744-2.545 12.197 12.197 0 0 0 2.531-3.757c.624-1.433.936-2.966.936-4.601 0-1.635-.312-3.168-.936-4.601a12.034 12.034 0 0 0-2.53-3.744A12.034 12.034 0 0 0 18.1 2.624C16.668 2 15.135 1.687 13.5 1.687c-1.635 0-3.168.313-4.601.937a12.034 12.034 0 0 0-3.744 2.53A12.034 12.034 0 0 0 2.624 8.9C2 10.332 1.687 11.865 1.687 13.5c0 1.635.313 3.168.937 4.601a12.197 12.197 0 0 0 2.53 3.757A11.921 11.921 0 0 0 8.9 24.403c1.433.624 2.966.936 4.601.936zm5.379-16.77a.834.834 0 0 1 .606-.263c.229 0 .431.088.607.263a.805.805 0 0 1 .237.594c0 .237-.079.435-.237.593L11.68 18.22l-.106.105a.765.765 0 0 1-.593.264.765.765 0 0 1-.593-.264L6.196 14.16a.805.805 0 0 1-.237-.593c0-.237.08-.435.237-.593a.834.834 0 0 1 .607-.264c.228 0 .43.088.606.264l3.56 3.56 7.91-7.964z'
}

export const Icon = ({ iconName, viewBox, color, ...other }) => {
    return (
        <svg
            {...svgStyle}
            viewBox={viewBox}
            {...other}
        >
            <path {...pathStyle(color)} d={icons[iconName]} />
        </svg>
    )
}

Icon.propTypes = {
    iconName: string.isRequired,
    viewBox: string,
    color: string
}

Icon.defaultProps = {
    viewBox: '0 0 24 24'
}
