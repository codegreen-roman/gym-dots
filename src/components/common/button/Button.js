import React from 'react'
import { func, node } from 'prop-types'
import glamorous from 'glamorous'
import { loginButtonStyles, linkButtonStyles, footerButtonStyles } from './Button.glamor'

const Button = ({ onClickAction, children, ...other }) => (
    <button {...other} onClick={onClickAction}>
        {children}
    </button>
)

Button.propTypes = {
    onClickAction: func.isRequired,
    children: node
}

const LinkButton = glamorous(Button)(linkButtonStyles)
const LoginButton = glamorous(Button)(loginButtonStyles)
const FooterButton = glamorous(Button)(footerButtonStyles)

export {
    Button,
    LoginButton,
    LinkButton,
    FooterButton
}
