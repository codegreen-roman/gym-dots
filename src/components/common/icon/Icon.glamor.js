import { css } from 'glamor'

export const svgStyle = css({
    display: 'inline-block',
    verticalAlign: 'middle'
})

export const pathStyle = color =>
    css({
        fill: color
    })
