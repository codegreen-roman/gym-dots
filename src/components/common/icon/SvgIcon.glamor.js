import glamorous from 'glamorous'
import { variables } from '../../../utils/variables'

export const StyledSvg = glamorous.svg(
    {
        display: 'inline-block',
        color: 'black',
        height: '24px',
        width: '24px',
        userSelect: 'none',
        verticalAlign: 'middle'
    },
    ({ fillColor }) => (
        fillColor
            ? { fill: fillColor }
            : { fill: variables.$color2 }
    ),
    ({ big }) => (
        big
            ? { width: '36px', height: '36px' }
            : { width: '24px', height: '24px' }
    )
)
