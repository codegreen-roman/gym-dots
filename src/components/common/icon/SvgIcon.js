import React from 'react'
import { string, node, bool } from 'prop-types'
import { curry, prop } from 'ramda'
import { StyledSvg } from './SvgIcon.glamor'
import { icons24, icons36 } from '../../../utils/icons'

/* Wrapper to create different svg icon components */
const SvgWrap = curry((path, props) => (
    <SvgIcon {...props}>
        <path d={path} />
    </SvgIcon>
))

const propOfIcons24 = key => prop(key, icons24)
const propOfIcons36 = key => prop(key, icons36)

/* 24 24 viewbox icons */
const MenuIcon = SvgWrap(propOfIcons24('menu'))
const FacebookIcon = SvgWrap(propOfIcons24('facebook'))
const TwitterIcon = SvgWrap(propOfIcons24('twitter'))
const GuestIcon = SvgWrap(propOfIcons24('guest'))

/* 36 36 viewbox icons */
const DumbbellsIcon = SvgWrap(propOfIcons36('dumbbells'))
const CheckIcon = SvgWrap(propOfIcons36('check'))

/* Base svg component */
const SvgIcon = ({ big, viewBox, children, ...other }) => (
    <StyledSvg {...other} big={big} viewBox={big ? '0 0 36 36' : viewBox}>
        {children}
    </StyledSvg>
)

SvgIcon.propTypes = {
    viewBox: string,
    big: bool,
    children: node
}

SvgIcon.defaultProps = {
    viewBox: '0 0 24 24',
    big: false
}

MenuIcon.displayName = 'MenuIcon'
FacebookIcon.displayName = 'FacebookIcon'
TwitterIcon.displayName = 'TwitterIcon'
GuestIcon.displayName = 'GuestIcon'
DumbbellsIcon.displayName = 'DumbbellsIcon'
CheckIcon.displayName = 'CheckIcon'

export {
    MenuIcon,
    FacebookIcon,
    TwitterIcon,
    GuestIcon,
    DumbbellsIcon,
    CheckIcon,
    SvgIcon,
    SvgWrap,
    propOfIcons24,
    propOfIcons36
}
