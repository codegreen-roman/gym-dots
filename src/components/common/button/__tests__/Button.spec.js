import React from 'react'
import { Button, LoginButton, LinkButton } from '../Button'
import { mount, shallow } from 'enzyme'
import { MenuIcon, FacebookIcon } from '../../icon/SvgIcon'

const defaultProps = {
    onClickAction: jest.fn(),
    children: null,
}

const setup = (props = defaultProps) => {
    return {
        mountWrapper: mount(<Button {...props} />),
        shallowWrapper: shallow(<Button {...props} />),
        props
    }
}

describe('Button', () => {

    it('should render as expected', () => {
        const { shallowWrapper } = setup()
        expect(shallowWrapper).toMatchSnapshot()
    })

    it('should invoke onClickAction', () => {
        const { mountWrapper } = setup()
        mountWrapper.simulate('click')
        expect(typeof mountWrapper.props().onClickAction === 'function').toBeTruthy()
        expect(mountWrapper.props().onClickAction).toHaveBeenCalled()
        expect(mountWrapper.props().onClickAction).toHaveBeenCalledTimes(1)
    })

    it('should spread attributes', () => {
        const mountWrapper = mount(<Button data-test='btn' disabled={false} {...defaultProps} />)
        expect(mountWrapper.props()['data-test']).toBe('btn')
        expect(mountWrapper.props().disabled).toBe(false)
    })

    describe('Login button', () => {
        it('should render login button with icon', () => {
            const mountWrapper = mount(<LoginButton {...defaultProps}><FacebookIcon /></LoginButton>)
            expect(mountWrapper).toMatchSnapshot()
        })

        it('should call action on login button click', () => {
            const mountWrapper = mount(<LoginButton {...defaultProps}><MenuIcon /></LoginButton>)
            mountWrapper.simulate('click')
            expect(mountWrapper.props().onClickAction).toHaveBeenCalled()
        })
    })

    describe('Link button', () => {
        it('should render linkbutton wrapping menu icon', () => {
            const mountWrapper = mount(<LinkButton {...defaultProps}><MenuIcon /></LinkButton>)
            expect(mountWrapper).toMatchSnapshot()
        })

        it('should call action on click', () => {
            const mountWrapper = mount(<LinkButton {...defaultProps}><MenuIcon /></LinkButton>)
            mountWrapper.simulate('click')
            expect(mountWrapper.props().onClickAction).toHaveBeenCalled()
        })
    })

})
