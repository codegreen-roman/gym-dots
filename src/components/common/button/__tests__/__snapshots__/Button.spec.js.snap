// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Link button should render linkbutton wrapping menu icon 1`] = `
.css-1fazozq,
[data-css-1fazozq] {
  display: inline-block;
  color: black;
  height: 24px;
  width: 24px;
  user-select: none;
  vertical-align: middle;
  fill: #5E6B7D;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

.css-vbolah,
[data-css-vbolah] {
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  font-size: 14px;
  line-height: 24px;
  font-weight: 500;
  text-align: center;
  color: #5E6B7D;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  height: 40px;
  justify-content: center;
  align-items: stretch;
  border: 0px;
  background: none;
  overflow: hidden;
  text-decoration: none;
  transition: all 300ms;
  outline: none;
  padding: 0px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-transition: all 300ms;
  -moz-transition: all 300ms;
}

<glamorous(Button)
  onClickAction={[Function]}
>
  <Button
    className="css-vbolah"
    onClickAction={[Function]}
  >
    <button
      className="css-vbolah"
      onClick={[Function]}
    >
      <MenuIcon>
        <SvgIcon
          big={false}
          viewBox="0 0 24 24"
        >
          <glamorous(svg)
            big={false}
            viewBox="0 0 24 24"
          >
            <svg
              className="css-1fazozq"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.881 12c0-.655-.537-1.185-1.2-1.185H6.6c-.663 0-1.2.53-1.2 1.185s.537 1.185 1.2 1.185h16.081c.663 0 1.2-.53 1.2-1.185zm-1.263 5.63H10.2c-.663 0-1.2.53-1.2 1.185C9 19.469 9.537 20 10.2 20h12.418c.662 0 1.2-.53 1.2-1.185s-.538-1.185-1.2-1.185zM22.8 6.37H1.2C.537 6.37 0 5.84 0 5.185 0 4.531.537 4 1.2 4h21.6c.663 0 1.2.53 1.2 1.185s-.537 1.185-1.2 1.185z"
              />
            </svg>
          </glamorous(svg)>
        </SvgIcon>
      </MenuIcon>
    </button>
  </Button>
</glamorous(Button)>
`;

exports[`Button Login button should render login button with icon 1`] = `
.css-fum0x2,
[data-css-fum0x2] {
  display: -webkit-inline-box;
  display: -moz-inline-box;
  display: -ms-inline-flexbox;
  display: -webkit-inline-flex;
  display: inline-flex;
  font-size: 14px;
  line-height: 24px;
  font-weight: 500;
  text-align: center;
  color: #5E6B7D;
  cursor: pointer;
  white-space: nowrap;
  text-overflow: ellipsis;
  height: 40px;
  justify-content: center;
  align-items: stretch;
  border: none;
  background: none;
  overflow: hidden;
  text-decoration: none;
  transition: all 300ms;
  outline: none;
  border-width: 1px;
  border-style: solid;
  border-color: #5E6B7D;
  border-radius: 2px;
  padding: 4px 16px 4px 8px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-transition: all 300ms;
  -moz-transition: all 300ms;
}

.css-1fazozq,
[data-css-1fazozq] {
  display: inline-block;
  color: black;
  height: 24px;
  width: 24px;
  user-select: none;
  vertical-align: middle;
  fill: #5E6B7D;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
}

<glamorous(Button)
  onClickAction={[Function]}
>
  <Button
    className="css-fum0x2"
    onClickAction={[Function]}
  >
    <button
      className="css-fum0x2"
      onClick={[Function]}
    >
      <FacebookIcon>
        <SvgIcon
          big={false}
          viewBox="0 0 24 24"
        >
          <glamorous(svg)
            big={false}
            viewBox="0 0 24 24"
          >
            <svg
              className="css-1fazozq"
              viewBox="0 0 24 24"
            >
              <path
                d="M13.491079,21.9911739 L13.491079,12.8737864 L16.4392523,12.8737864 L16.8810535,9.3168579 L13.491079,9.3168579 L13.491079,7.04854369 C13.491079,6.01588703 13.7629567,5.31862312 15.1903144,5.31862312 L17,5.31862312 L17,2.14121801 C16.6856415,2.09708738 15.606627,2 14.357689,2 C11.7408666,2 9.9566695,3.65931156 9.9566695,6.69549868 L9.9566695,9.3168579 L7,9.3168579 L7,12.8737864 L9.9566695,12.8737864 L9.9566695,22 L13.491079,22 L13.491079,21.9911739 Z"
              />
            </svg>
          </glamorous(svg)>
        </SvgIcon>
      </FacebookIcon>
    </button>
  </Button>
</glamorous(Button)>
`;

exports[`Button should render as expected 1`] = `
<button
  onClick={[Function]}
/>
`;
